src/app/search/basic-search/facet-search.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { Observable } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+import { Base } from '../../shared/base';
+import { SearchQuery } from '../state/search.query';
+import { SearchService } from '../state/search.service';
+
+@Component({
+  selector: 'app-facet-search',
+  templateUrl: './facet-search.component.html',
+  styleUrls: ['./facet-search.component.scss'],
+})
+export class FacetSearchComponent extends Base implements OnInit {
```

Comment:
Haven't actually run this, but this looks really cool...

Just skimming the code, I feel like with a little more work it could be more generic, and not tied directly to author, like on one 20 -- we would eventually want to use this component for other facets as well.

But I don't know this code very well, so it might be trickier to do than say. Also, you could genericize it in the next iteration, when you use it for a different component.